uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -1192,52 +1192,36 @@ sub computeSNR {
     $minc_file_arr->execute($tarchiveID);
 
     while ($row = $minc_file_arr->fetchrow_hashref()) {
-           $file = $row->{'file'};
-           $fileID = $row->{'FileID'};
-           $base = basename($file);
-           $fullpath = $data_dir . "/" . $file;
-           if (defined(&Settings::getSNRModalities)
-                && Settings::getSNRModalities($base)) {
-               $cmd = "noise_estimate --snr $fullpath";
-               $SNR = `$cmd`;
-               $SNR =~ s/\n//g;
-               print "$cmd \n" if ($this->{verbose});
-               print "SNR is: $SNR \n" if ($this->{verbose});
-
-                ### Update the SNR in files table
-                $query = "SELECT ParameterTypeID from parameter_type pt ";
-		 $where = "WHERE pt.Name='SNR'";
-		 $query = $query . $where;
-
-		if ($this->{debug}) {
-		    print $query . "\n";
-		}
-
-		$sth = ${$this->{'dbhr'}}->prepare($query);
-		$sth->execute();
-		if ( $sth->rows > 0 ) {
-		    $paramID = $sth->fetchrow_array;
- 		}
-
-                $query = "INSERT INTO parameter_file SET Value=?, ".
-			 "FileID=?, ParameterTypeID=?";
-                if ($this->{debug}) {
-                    print $query . "\n";
+        $filename = $row->{'file'};
+        $fileID = $row->{'FileID'};
+        $base = basename($filename);
+        $fullpath = $data_dir . "/" . $filename;
+        if (defined(&Settings::getSNRModalities)
+            && Settings::getSNRModalities($base)) {
+                $cmd = "noise_estimate --snr $fullpath";
+                $SNR = `$cmd`;
+                $SNR =~ s/\n//g;
+                print "$cmd \n" if ($this->{verbose});
+                print "SNR is: $SNR \n" if ($this->{verbose});
+                my $file = NeuroDB::File->new($this->{dbhr});
+                $file->loadFile($fileID);
+                $SNR_old = $file->getParameter('SNR');
+                if ((defined ($SNR_old)) && ($SNR_old != $SNR)) {
```

Comment:
SNR is:  
Argument "" isn't numeric in numeric ne (!=) at /data/ibis/bin/mri/uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm 

May need to break it into two if.